src/node_http2.cc

```diff
@@ -1769,20 +1765,20 @@ void Http2Session::OnStreamRead(ssize_t nread, const uv_buf_t& buf_) {
     memcpy(new_buf.data(), stream_buf_.base + stream_buf_offset_, pending_len);
     memcpy(new_buf.data() + pending_len, buf.data(), nread);
 
-    // The data in stream_buf_ is already accounted for, add nread received
-    // bytes to session memory but remove the already processed
-    // stream_buf_offset_ bytes.
-    // TODO(@jasnell): There are some cases where nread is < stream_buf_offset_
-    // here but things still work. Those need to be investigated.
-    // CHECK_GE(nread, stream_buf_offset_);
-    IncrementCurrentSessionMemory(nread - stream_buf_offset_);
-
     buf = std::move(new_buf);
     nread = buf.size();
     stream_buf_offset_ = 0;
     stream_buf_ab_.Reset();
+
+    // We have now fully processed the stream_buf_ input chunk (by moving the
+    // remaining part into buf, which will be accounted for below).
+    DecrementCurrentSessionMemory(stream_buf_.len);
```

Comment:
I guess in my PR this should've been 
`IncrementCurrentSessionMemory(nread - stream_buf_.len);`

After rebasing on https://github.com/nodejs/node/pull/30684 I didn't pay attention to the fact that we actually skip accounting for the *whole* buffer, not just the remainder (after `stream_buf_offset_` part which is what I accounted for) hence the error most likely.

Reply:
Did you want to submit a different PR that doesn't revert but instead fixes this issue? Or should I go ahead and land this?

Reply:
I'll submit a PR in a bit though the only way to test flakiness for me right now is via our CI so that may take some time. Feel free to land this and I'll reapply the changes in a new PR if needed and it actually works =).

Reply:
Ok,  guess it's not that easy :disappointed:, with the change it just fails.
Funny thing

`    IncrementCurrentSessionMemory(nread - stream_buf_.len);` fails

-----------------------------------
But this works
```c++
if (LIKELY(stream_buf_offset_ == 0)) {
    buf.Resize(nread);
} else {
  ...
  DecrementCurrentSessionMemory(stream_buf_.len);
}
IncrementCurrentSessionMemory(nread);
```

Aren't they the same thing?